docs/developing.rst

```diff
@@ -0,0 +1,224 @@
+
+with ``TEST`` your test function's name.
+This call basically adds code coverage reports to account for the new test, and runs the actual
+test in addition.
+
+(6) Using the five-echo set as a template, you should then edit ``.circlec/config.yml`` to add your
+test, calling the same name you define in the ``Makefile``.
+
+Viewing CircleCI Outputs
+------------------------
+If you need to take a look at a failed test on CircleCI rather than locally, you can use the
+following block to retrieve artifacts (see CircleCI documentation here_)
+
+.. code-block:: bash
+
+    export CIRCLE_TOKEN=':your_token'
+
+    curl https://circleci.com/api/v1.1/project/:vcs-type/:username/:project/$build_number/artifacts?circle-token=$CIRCLE_TOKEN \
+       | grep -o 'https://[^"]*' \
+       | sed -e "s/$/?circle-token=$CIRCLE_TOKEN/" \
+       | wget -v -i -
+
+To get a CircleCI token, follow the instructions for `getting one`_.
+You cannot do this unless you are part of the ME-ICA/tedana organization.
+If you don't want all of the artifacts, you can go to the test details and use the browser to
+manually select the files you would like.
+
+Worked Example
+==============
+Suppose we want to add a function in ``tedana`` that creates a file called ```hello_world.txt`` to
+be stored along the outputs of the ``tedana`` workflow.
+
+First, we merge the repository's ``master`` branch into our own to make sure we're up to date, and
+then we make a new branch called something like ``feature/say_hello``.
+Any changes we make will stay on this branch.
+We make the new function and call it ``say_hello`` and locate this function inside of ``io.py``.
+We'll also need to make a unit test.
+(Some developers actually make the unit test before the new function; this is a great way to make
+sure you don't forget to create it!)
+Since the function lives in ``io.py``, its unit test should go into ``test_io.py``.
+The job of this test is exclusively to tell if the function we wrote does what it claims to do
+without errors.
+So, we define a new function in ``test_io.py`` that looks something like this:
+
+.. code-block:: python
+
+    def test_say_hello():
+        # run the function
+        say_hello()
+        # test the function
+        assert op.exists('hello_world.txt')
+        # clean up
+        os.remove('hello_world.txt')
+
+We should see that our unit test is successful via
+
+.. code-block:: bash
+
+    pytest $TEDANADIR/tedana/tests/test_io.py -k test_say_hello
+    
+If not, we should continue editing the function until it passes our test.
+Let's suppose that suddenly, you realize that what would be even more useful is a function that
+takes an argument, ``place``, so that the output filename is actually ``hello_PLACE``, with
+``PLACE`` the value passed and ``'world'`` as the default value.
+We merge any changes from the upstream master branch into our branch via
+
+.. code-block:: bash
+
+    git checkout feature/say_hello  # unless you're already there
```

Comment:
```suggestion
    git checkout feature/say_hello
```

Reply:
Hm, I did this because @emdupre had pointed that it's unclear that you only have to do this if you've left the branch. Do you think this adds to the confusion instead of clarifying?